commit e61f234079b49c7e075b12551797fc4954704019
Author: Jouni Malinen <jouni.malinen@atheros.com>
Date:   Fri Apr 24 00:00:00 2009 +0300

    nl80211: Send timeout event on failed direct probe
    
    If the direct probe times out, we need to send the authentication
    timeout event to notify SME in the same way as we notify on timeout
    with authentication frames since the direct probe is run as part of
    the authentication attempt.
    
    Signed-off-by: Jouni Malinen <jouni.malinen@atheros.com>
    Signed-off-by: John W. Linville <linville@tuxdriver.com>

diff --git a/net/mac80211/mlme.c b/net/mac80211/mlme.c
index 42f33fd3c5ec..30a0034b9124 100644
--- a/net/mac80211/mlme.c
+++ b/net/mac80211/mlme.c
@@ -890,7 +890,7 @@ static void ieee80211_direct_probe(struct ieee80211_sub_if_data *sdata)
 		printk(KERN_DEBUG "%s: direct probe to AP %pM timed out\n",
 		       sdata->dev->name, ifmgd->bssid);
 		ifmgd->state = IEEE80211_STA_MLME_DISABLED;
-		ieee80211_sta_send_apinfo(sdata);
+		cfg80211_send_auth_timeout(sdata->dev, ifmgd->bssid);
 
 		/*
 		 * Most likely AP is not in the range so remove the